### PR TITLE
Revert missing docs in Merkle DB.

### PR DESCRIPTION
**Pull request by pavel-mukhanov**

### CODE REVIEW

File: components/merkledb/src/proof_map_index/node.rs [hunk 1]
```diff
@@ -26,57 +26,61 @@ use crate::{BinaryKey, BinaryValue, HashTag, ObjectHash};
 
 const BRANCH_NODE_SIZE: usize = 2 * (HASH_SIZE + PROOF_PATH_SIZE);
 
+/// Enum that represents type of the ProofMap node.
```
Comment (Maria-Nosyk) [inline components/merkledb/src/proof_map_index/node.rs]: represents the type

File: components/merkledb/src/proof_map_index/node.rs [hunk 2]
```diff
@@ -26,57 +26,61 @@ use crate::{BinaryKey, BinaryValue, HashTag, ObjectHash};
...
 
+/// Struct that encapsulates raw representation of the branch node.
 #[derive(Clone, PartialEq)]
 pub struct BranchNode {
     raw: Vec<u8>,
 }
 
 impl BranchNode {
-    pub fn empty() -> Self {
+    pub(crate) fn empty() -> Self {
```
Comment (aleksuss) [inline components/merkledb/src/proof_map_index/node.rs]: This method is used in the benchmarks. So it should be public.

File: components/merkledb/src/proof_map_index/node.rs (hunk 1 above)
Comment (aleksuss) [inline components/merkledb/src/proof_map_index/node.rs]:
```suggestion
/// Enum that represents type of the `ProofMap` node.
```

File: components/merkledb/src/proof_map_index/node.rs [hunk 3]
```diff
@@ -26,57 +26,61 @@ use crate::{BinaryKey, BinaryValue, HashTag, ObjectHash};
 
 const BRANCH_NODE_SIZE: usize = 2 * (HASH_SIZE + PROOF_PATH_SIZE);
 
+/// Enum that represents type of the ProofMap node.
 #[derive(Debug)]
 pub enum Node {
+    /// Leaf node, contains hash of the leaf value.
```
Comment (Maria-Nosyk) [inline components/merkledb/src/proof_map_index/node.rs]: Leaf node that contains a hash of the leaf value.

File: components/merkledb/src/proof_map_index/node.rs [hunk 4]
```diff
@@ -26,57 +26,61 @@ use crate::{BinaryKey, BinaryValue, HashTag, ObjectHash};
...
     Leaf(Hash),
+    /// Branch node.
     Branch(BranchNode),
 }
 
+/// Struct that encapsulates raw representation of the branch node.
```
Comment (Maria-Nosyk) [inline components/merkledb/src/proof_map_index/node.rs]: Structure that encapsulates a raw representation of the branch node.

**Pull request merged**